src/python/pants/binaries/binary_tool.py

```diff
@@ -230,3 +276,28 @@ def tar_xz_extractor(self):
 
   def _executable_location(self):
     return os.path.join(self.select(), 'bin', 'xz')
+
+
+@dataclass(frozen=True)
+class BinaryToolFetchRequest:
+  tool: BinaryToolBase
+
+
+@rule
+async def fetch_binary_tool(req: BinaryToolFetchRequest) -> Snapshot:
```

Comment:
It doesn't feel like breaking this out as a separate rule is worthwhile: would inline it into `fetch_binary_tool_single_file_exe`.

Reply:
Ok so I would, but some `BinaryTool`s don't output to an executable, rather they download a tgz or zip archive. I thought that splitting this out would make it easier to add those capabilities in v2 later.

Reply:
Hm, ok! I think I agree with this approach in general, but not every binary tool will necessarily be an executable, though -- some could be a resource file, or possibly a compressed archive. There's a (hacky, non-hermetic) `unzip` command usage in #8940 -- if we can land this PR, we could then also package the `unzip` tool as a `NativeTool`, and then we could get to feature parity with the v1 version of `BinaryToolBase` -- being able to automatically decompress and snapshot an archive.

In the vein of your next comment, though, it seems like it might be appropriate to put more logic in this rule anyway (instead of creating the private `_get_url_to_fetch()` method). That still allows for other rules to just request a `Snapshot`, if their chosen tool isn't actually just a single file.

If we're just looking for a wrapper class so we're not tossing around raw `Snapshot`s, I can totally do that too. It might make it easier to extend later as well!